spring-data-neo4j/src/main/java/org/springframework/data/neo4j/repository/config/Neo4jMappingContextFactoryBean.java

```diff
@@ -27,12 +27,35 @@
  *
  * @author Mark Angrish
  * @author Nicolas Mervaillie
+ * @author Michael J. Simons
  */
 public class Neo4jMappingContextFactoryBean extends AbstractFactoryBean<Neo4jMappingContext> implements
 		ApplicationContextAware {
 
+	private final String sessionFactoryBeanName;
+
 	private ListableBeanFactory beanFactory;
 
+	/**
+	 * Uses the default session factory name {@link Neo4jRepositoryConfigurationExtension#DEFAULT_SESSION_FACTORY_BEAN_NAME}
+	 * for finding the session factory passed to the {@link Neo4jMappingContext}.
+	 *
+	 * @deprecated
+	 */
+	public Neo4jMappingContextFactoryBean() {
+		this(Neo4jRepositoryConfigurationExtension.DEFAULT_SESSION_FACTORY_BEAN_NAME);
+	}
+
+	/**
+	 * Configures the mapping context with a named {@link SessionFactory}.
+	 *
+	 * @param sessionFactoryBeanName
+	 * @since 5.1.0
+	 */
+	Neo4jMappingContextFactoryBean(String sessionFactoryBeanName) {
+		this.sessionFactoryBeanName = sessionFactoryBeanName;
```

Comment:
How about adding `Assert.notEmpty(…)` for `sessionFactoryBeanName`? Makes also sense to document the constraint in the javadoc. Mind also turning the constructor into a `public` one so the class can be instantiated from outside the package?